libs/community/langchain_community/tools/vectorstore/tool.py

```diff
@@ -51,9 +54,28 @@ def _run(
         chain = RetrievalQA.from_chain_type(
             self.llm, retriever=self.vectorstore.as_retriever()
         )
-        return chain.run(
-            query, callbacks=run_manager.get_child() if run_manager else None
+        return chain.invoke(
```

Comment:
change fron `run` to `invoke` as the former is deprecated.

Reply:
@cbornet 

This is incorrect fix. I get


`
AttributeError("'CallbackManager' object has no attribute 'ignore_chain'")
`
as root cause and
`
AttributeError: 'CallbackManager' object has no attribute 'raise_error'
`

The reason is that this code puts run_manager into **handlers**.

Please revert this change back. 

if you want to change from call chain.run() to chain.invoke() this is fine, but the code should be correect.

Reply:
@cbornet 
I've opened an issue https://github.com/langchain-ai/langchain/issues/18460